File: Check_if_Array_Is_Sorted_and_Rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        int small=0;
        for(int i=1 ; i<nums.size() ;i++){
            if(nums[small]>=nums[i]){
                small=i;
            }
        }
        for(int i = 0 ;i<nums.size()-1;i++){
            int index = (small + i)%nums.size();
            if(nums[index]>nums[(index+1)%nums.size()]){
                return false;
            } 
            
        }
        return true;  
    }
};

File: Check_if_Array_Is_Sorted_and_Rotated.java
class Solution {
    public boolean check(int[] nums) {
        // int s=0;
        // int count=0;
        // for(int i=0 ; i<nums.length ;i++){
        //     if(nums[i]>nums[(i+1)%nums.length]){
        //         s=nums[i];
        //         count++;
        //     }
        // }

        // int max = Integer.MIN_VALUE;
        // for(int i=0 ; i<nums.length ;i++){
        //     if(nums[i]>max){
        //         max=nums[i];
        //     }
        // }
        // if(count>=2){
        //     return false;
        // }

        // if(max==s || s==0){
        //     return true;
        // }else{
        //     return false;
        // }
        int count =0;
        int n = nums.length;
        for(int i=0 ; i<n ; i++){
            if(nums[(i%n)]>nums[(i+1)%n]){
                count++;
                if(count>1){
                    return false;
                }
            }
        }
            return true;
    }
}